### PR TITLE
fix: [FFM-4345]: update form fields to use TagInput only when 'in' operator selected

### PR DESCRIPTION
**Pull request by Kira-Jennings**
- Form fields in section 'Target Based on Condition' have been updated to allow multiple values only when the 'in' operator has been selected
- updated unit tests

image
image

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 3 Issues 
* Code issue 3 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10799)

**Review by crystal-fong — DISMISSED** [review]

File: src/modules/75-cf/pages/target-group-detail/components/EditTargetGroupCriteria/__tests__/RuleRow.test.tsx [hunk 1]
```diff
@@ -81,9 +90,74 @@ describe('RuleRow', () => {
     })
   })
 
-  test('it should display a tag input for the value', async () => {
+  test('it should display a text input for the value when no op selected', async () => {
     renderComponent()
-    expect(screen.getByLabelText('cf.segmentDetail.values')).toBeInTheDocument()
+
+    const ruleRow = screen.getByTestId('ruleRowFields')
```
Comment (laura-mannus) [inline src/modules/75-cf/pages/target-group-detail/components/EditTargetGroupCriteria/__tests__/RuleRow.test.tsx:96]: you can move this outside the tests and declare it only once so you don't have to remake it in every test

Comment (Kira-Jennings) [reply src/modules/75-cf/pages/target-group-detail/components/EditTargetGroupCriteria/__tests__/RuleRow.test.tsx:96]: That's usually the case but in this scenario that would mean declaring it outside of the scope of the test, and therefore before anything is rendered which results in ruleRow being undefined
If we didn't need to send a different operator every time we're calling `renderComponent` I could have called the render within a beforeEach, then declared `ruleRow` outside of the tests with no problems

Comment (Kira-Jennings) [reply src/modules/75-cf/pages/target-group-detail/components/EditTargetGroupCriteria/__tests__/RuleRow.test.tsx:96]: image
image

Comment (Kira-Jennings) [reply src/modules/75-cf/pages/target-group-detail/components/EditTargetGroupCriteria/__tests__/RuleRow.test.tsx:92]: I've removed all the `waitFor` blocks in this file that weren't needed, only the one in last test is required

**Review by laura-mannus — APPROVED** [review]
nice

**Review by crystal-fong — APPROVED** [review]

**Review by crystal-fong — APPROVED** [review]

**Pull request merged**